src/cs/Ssh/TaskChain.cs

```diff
@@ -49,6 +49,16 @@ public async Task RunInSequence(Func<Task> task, Action<Exception> onError, Func
 		try
 		{
 			await semaphore.WaitAsync(cancellation).ConfigureAwait(false);
+			
+			if (runInSequenceTask != null && 
```

Comment:
I think I see another issue. If `preTask` fails on line 112, then `completionSource` is never set, and the task chain will be blocked waiting for it forever. 

Not blocking his PR.

Reply:
`completionSource.SetResult` is done in finally block (line 122) right? so, should get executed even if `preTask` fails.

Reply:
Right. I missed it because GH hidden it and I didn't expand